### PR TITLE
PR blaster outer: Properly handle travis excludes.

### PR DESCRIPTION
**Pull request by jrafanie**
tldr; we weren't handling the exclusions of the older ruby test runs for ui-classic, nuage, and some other repos so we'd have to update the rubies, then they'd have to go and fix the excludes after the fact. 

Use YAML to parse and dump the file instead of reading the yaml as a string and
using gsub to replace values for a few reasons:

* the blind gsub across the whole yaml string could not replace the travis excluded ruby suites
* it could gsub in the wrong places
* it could not handle if we're changing the number of rubies we want to test

Note, we can't output comments we had in our .travis.yml so any comments will
now be lost.  We don't currently have any comments so it's not a problem.

We now only update the toplevel 'rvm' key and ['matrix']['exclude'] if they
exist and populated, instead of replacing all similar looking version strings.

Finally, we now force sort of rubies in ascending order to match existing
patterns in our files.

### CODE REVIEW

**Comment by jrafanie** [comment]
for example, old versions of this script would skip the excludes part of this PR:  https://github.com/ManageIQ/manageiq-ui-classic/pull/6336/files
